processor/spanmetricsprocessor/processor.go

```diff
@@ -260,31 +290,21 @@ func (p *processorImp) tracesToMetrics(ctx context.Context, traces ptrace.Traces
 	// This component no longer needs to read the metrics once built, so it is safe to unlock.
 	p.lock.Unlock()
 
-	if err != nil {
-		return err
-	}
-
-	if err = p.metricsExporter.ConsumeMetrics(ctx, m); err != nil {
-		return err
+	if err := p.metricsExporter.ConsumeMetrics(ctx, m); err != nil {
+		p.logger.Error("Failed ConsumeMetrics: " + err.Error())
```

Comment:
Use structured logs to make it easier to search in logging platforms:

```suggestion
		p.logger.Error("Failed ConsumeMetrics", zap.Error(err))
```

Reply:
Thanks, addressed in: [261ebbb](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17307/commits/261ebbb816f031a9f4cfdd5a432a87a7f40a53e9)